src/api/spec/helpers/webui/webui_helper_spec.rb

```diff
@@ -319,7 +319,69 @@
   end
 
   describe '#user_and_role' do
-    skip('Please add some tests')
+    let(:user) { create(:user) }
+    let(:nobody) { create(:user_nobody) }
+
+    before do
+      User.current = user
+    end
+
+    context 'for users' do
+      it 'display the users realname with its login' do
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user.realname}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'falls back to users login if realname is empty' do
+        user.realname = ''
+        user.save!
+
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.login}</a>"
+        )
+      end
+    end
+
+    context 'with options' do
+      it 'does not show an icon if option disables it' do
+        expect(user_and_role(user.login, nil, no_icon: true)).to eq(
+          "<a href=\"/user/show/#{user.login}\">#{user.realname} (#{user.login})</a>"
+        )
+      end
+
+      it 'only shows user login if short option is set' do
+        user.realname = ''
+        user.save!
+
+        expect(user_and_role(user.login, nil, short: true)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user}\" src=\"/user/icon/#{user.login}?size=20\" " + \
+          "/><a href=\"/user/show/#{user.login}\">#{user.login}</a>"
+        )
+      end
+    end
+
+    context 'for guests' do
+      it 'does not link to user profiles if current user is nobody' do
+        User.current = nobody
+
+        expect(user_and_role(user.login)).to eq(
+          "<img width=\"20\" height=\"20\" alt=\"#{user.realname}\" " + \
+          "src=\"/user/icon/#{user.login}?size=20\" />#{user.realname} (#{user.login})"
+        )
+      end
+    end
+
+    context 'with roles' do
```

Comment:
You don't display nobody. You display `user`. Nobody is only the current user. It doesn't matter what you'll set here. If'd set it to a logged in user, the method #user_and_role will just link to profiles then. If he's a guest, it would not link to anything.